build.gradle

```diff
@@ -112,28 +113,86 @@ dependencies {
     testImplementation "org.hamcrest:hamcrest:${hamcrest}"
 }
 
-test {
-    testLogging {
-        showStandardStreams = true // disable stdout & stderr redirection
-    }
-    useJUnitPlatform()
+// install the CLI, either from source code directly or from a GH release, before starting tests
+task runInstallForTesting(type: Exec){
+    boolean testInstallFromGitHub = project.hasProperty("testInstallFromGitHub");
+    String installMode = testInstallFromGitHub ? "GITHUB_RELEASE" : "SOURCE_CODE"
+    commandLine "sh", "./tools/install-for-testing.sh", installMode
+
+    dependsOn installDist
 }
 
-task unitTest(type: Test) {
-    useJUnitPlatform {
-        includeTags 'unit'
+// run tests with the specified tag and install mode.
+// 1) run unit tests directly against the source code
+//    ./gradlew runTestsWithTag -PtestTag=unit
+// 2) run integration tests against an installation built from source code
+//    ./gradlew runTestsWithTag -PtestTag=integration
+// 3) run integration tests against an installation built from the latest GitHub release
+//    ./gradlew runTestsWithTag -PtestTag=integration -PtestInstallFromGitHub
+task runTestsWithTag(type: Test) {
```

Comment:
There was enough shared that it felt like too much repetition, so I combined them into a single task. I tried having a setup task that the unit and integration test tasks called, but couldn't get that to work reliably in the right order. Agreed this could get unwieldy if we have a new third type of test. But I think I'm going to punt on reorganizing this until we get there.